app/services/activities/import_from_csv.rb

```diff
@@ -131,15 +133,18 @@ def create
         @activity.cache_roda_identifier
 
         implementing_organisation_builder = ImplementingOrganisationBuilder.new(@activity, row)
-        @activity.implementing_organisations = [implementing_organisation_builder.build]
+        if row["Implementing organisation name"].present? || row["Implementing organisation sector"].present?
+          implementing_organisation = implementing_organisation_builder.build
+          @activity.implementing_organisations = [implementing_organisation]
+        end
 
         return if @activity.save(context: Activity::VALIDATION_STEPS)
 
         @activity.errors.each do |attr_name, message|
           @errors[attr_name] ||= [@converter.raw(attr_name), message]
         end
 
-        implementing_organisation_builder.add_errors(@errors)
+        implementing_organisation_builder.add_errors(@errors) if @activity.implementing_organisations.any?
```

Comment:
I'm guessing this is the latest change you committed. Well spotted! I hadn't noticed that without that conditional things wouldn't work as expected. 
Happy for this PR to get merged :-)